.github/workflows/staging-deployment.yml

```diff
@@ -124,133 +109,56 @@ jobs:
-          
-          # Start Artemis-Service on node
-          echo "[INFO] Start artemis.service ..."
-          $SSH sudo systemctl start artemis
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: '3.x'
 
-      - name: Verify Primary Node Deployment
-        id: verify_deployment
-        timeout-minutes: 10
+      - name: Install Ansible
         run: |
-          while true; do
-            echo "Performing health check..."
-          
-            RESPONSE=$(curl -s -f $HEALTH_CHECK_URL || echo '{"status":"DOWN"}')
-            STATUS=$(echo $RESPONSE | grep -o '"status":"[^"]*"' | cut -d'"' -f4)
-          
-            if [ "$STATUS" = "UP" ]; then
-              echo "Health check passed! Application is UP"
-              exit 0
-            else
-              echo "Health check failed. Status: $STATUS"
-              echo "Waiting 10 seconds before next attempt..."
-              sleep 10
-            fi
-          done
+          python -m pip install --upgrade pip
+          pip install ansible
 
-      - name: Phase 2 - Deploy to Secondary Nodes
+      - name: Install Vault
         run: |
-          HOSTS_SPACE_SEPARATED=$(echo "$DEPLOYMENT_HOSTS_SECONDARY" | tr -d '\r' | tr '\n' ' ' | awk '{$1=$1};1')
-          WAR_FILE=$(ls -1 artifacts/*.war | head -n 1)
-          
-          echo "Debug: Hosts list: $HOSTS_SPACE_SEPARATED"
-
-          for node in $HOSTS_SPACE_SEPARATED
-          do
-            echo "##################################################################################################"
-            echo "[INFO] Deploy on $node ..."
-            echo "##################################################################################################"
-
-            # Build SSH-command
-            SSH="ssh -o LogLevel=ERROR -i ~/.ssh/id_rsa -l $DEPLOYMENT_USER $node"
+          wget https://releases.hashicorp.com/vault/1.8.1/vault_1.8.1_linux_amd64.zip
+          unzip vault_1.8.1_linux_amd64.zip
+          sudo mv vault /usr/local/bin/
+          vault --version
 
-            # Remove old artemis.war
-            echo "[INFO] Remove old artemis.war ..."
-            $SSH "rm -f /opt/artemis/artemis.war.old"
+      - name: Install Artemis-Ansible dependencies
+        run: |
+          pip3 install -r requirements.txt
+          ansible-galaxy collection install -r requirements.yml --force
+          ansible-galaxy install -r requirements.yml --force
+          ansible-galaxy install -r ~/.ansible/collections/ansible_collections/ls1intum/artemis/requirements.yml
 
-            # Copy new artemis.war to node
-            echo "[INFO] Copy new artemis.war ..."
-            scp -i ~/.ssh/id_rsa "$WAR_FILE" "$DEPLOYMENT_USER@$node:/opt/artemis/artemis.war.new"
+      - name: Set Vault token
+        run: |
+          echo "${{ secrets.VAULT_TOKEN }}" > ~/.vault-token
+          export VAULT_TOKEN=$(cat ~/.vault-token)
```

Comment:
Would it make sense to set `VAULT_ADDR` as well?

Reply:
The ansible.cfg file in the artemis-ansible repo already configures this, so this is not needed.

Reply:
True 👍